pallets/phala/src/compute/stake_pool_v2.rs

```diff
@@ -802,6 +802,67 @@ pub mod pallet {
 			Ok(())
 		}
 
+		#[pallet::weight(0)]
+		#[frame_support::transactional]	
+		pub fn reset_iter_pos(origin: OriginFor<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			base_pool::Pallet::<T>::ensure_migration_root(who)?;
+			StakepoolIterateStartPos::<T>::put(None::<u64>);
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		#[frame_support::transactional]
+		pub fn migrate_stakepools(origin: OriginFor<T>, max_iterations: u32) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			base_pool::Pallet::<T>::ensure_migration_root(who)?;
+			let mut last_pid = StakepoolIterateStartPos::<T>::get();
+			let mut iter = match last_pid {
+				Some(pid) => {
+					let key: Vec<u8> = base_pool::pallet::Pools::<T>::hashed_key_for(pid);
+					base_pool::pallet::Pools::<T>::iter_from(key)
+				}
+				None => base_pool::pallet::Pools::<T>::iter(),
+			};
+			let asset_account = <T as wrapped_balances::Config>::WPhaAssetId::get();
+			let mut i = 0;
+			for (pid, pool_proxy) in iter.by_ref() {
+				match pool_proxy {
+					PoolProxy::Vault(_) => continue,
+					PoolProxy::StakePool(pool_info) => {
+						let mut total_lock = Zero::zero();
+						pool_info.workers.into_iter().for_each(|pubkey| {
+							let session: T::AccountId = pool_sub_account(pid, &pubkey);
+							total_lock += computation::Stakes::<T>::get(&session).unwrap_or_default();
+						});
+						pool_info.cd_workers.into_iter().for_each(|pubkey| {
+							let session: T::AccountId = pool_sub_account(pid, &pubkey);
+							total_lock += computation::Stakes::<T>::get(&session).unwrap_or_default();
+						});
+						let curr_lock: BalanceOf<T> =
+						<pallet_assets::pallet::Pallet<T> as Inspect<T::AccountId>>::balance(
+							asset_account,
+							&pool_info.lock_account,
+						);
+						if curr_lock < total_lock {
```

Comment:
Is it possible curr_lock > total lock? For security, I suggest to either return an error here, or we can simulate it offline first.

Reply:
don e